app/views/layouts/application.html.erb

```diff
@@ -0,0 +1,64 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+
+    <title><%= content_for?(:title) ? yield(:title) : "Untitled" %></title>
+
+    <%= stylesheet_link_tag  "application" %>
+    <%= javascript_include_tag "application", 'data-turbolinks-track' => true %>
+    <%= csrf_meta_tags %>
+  </head>
+  <nav>
+  <%=link_to "Home", root_path ,class:"button"%>
+<%if user = User.find_by(id: session[:user_id]) %>
+  <span id="user-signed-in">Signed in as: <span id="user-name"><%=user.username%></span> </span><%=link_to "Log out", signout_path, class:"button"%>
+<%else%>
+  <%= link_to "Sign-in with Github", "/auth/github", class:"button"  %>
+<%end%>
+  </nav>
+
+  <%if params[:query].nil?%>
```

Comment:
Maybe simply put `main` elements with these ids in the views for homepage and search etc.